.github/actions/setup-lxd-cluster/action.yml

```
@@ -0,0 +1,60 @@
+name: LXD Cluster Setup
+description: Sets up the LXD cluster for testing.
+
+inputs:
+  cluster-name:
+    description: |
+      The name of the cluster. Used as a prefix for cluster member
+      instances and as a name of the remote on a local LXD client.
+    default: cls
+  cluster-size:
+    description: The size of the cluster.
+    default: 3
+  instance-type:
+    description: The instance type of the cluster member.
+    default: 'container'
+    options:
+      - container
+      - virtual-machine
+  lxd-snap-channel:
+    description: Snap channel to use for installing LXD on each cluster member.
+    default: latest/edge
+  storage-driver:
+    description: The storage driver to use for default storage pool.
+    default: dir
+    options:
+      - dir
+      - zfs
+
+runs:
+  using: composite
+  steps:
+      # Remove Docker to avoid conflicts with LXD networks.
+    - name: Remove Docker
+      uses: canonical/lxd/.github/actions/disable-docker@main
+
+    - name: Remove existing snaps
+      shell: bash
+      run: |
+        for s in $(snap list | awk '!/^(Name|core|snapd)/ {print $1}'); do
+          sudo snap remove --purge "${s}" || true
+        done
+
+    - name: Setup LXD
+      shell: bash
+      run: |
+        sudo snap install lxd --channel=latest/stable
+        sudo lxd waitready --timeout 60
+        sudo lxd init --auto --network-port=8443 --network-address=localhost
+        sudo chmod 777 /var/snap/lxd/common/lxd/unix.socket
```

We have a long term goal of closing `latest/stable` and there is also `canonical/setup-lxd@main` that might work for this?

```suggestion
   - uses: canonical/setup-lxd@main
     with:
       preseed: |
         config:
           core.https_address: 127.0.0.1:8443
```